nowcasting_dataset/dataset/datasets.py

```diff
@@ -338,3 +387,93 @@ def worker_init_fn(worker_id):
         # The NowcastingDataset copy in this worker process.
         dataset_obj = worker_info.dataset
         dataset_obj.per_worker_init(worker_info.id)
+
+
+def subselect_data(
+    batch: example.Example,
+    required_keys: Union[Tuple[str], List[str]],
+    current_timestep_index: int,
+    history_minutes: int,
+    forecast_minutes: int,
+) -> example.Example:
+    """
+    Subselects the data temporally. This function selects all data within the range [history_minutes, forecast_minutes]
+
+    Args:
+        batch: Example dictionary containing at least the required_keys
+        required_keys: The required keys present in the dictionary to use
+        current_timestep_index: The index into either SATELLITE_DATETIME_INDEX or NWP_TARGET_TIME giving the current timestep
+        history_minutes: How many minutes of history to use
+        forecast_minutes: How many minutes of future data to use for forecasting
+
+    Returns:
+        Example with only data between [history_minutes, forecast_minutes] remaining
+    """
+    # We are subsetting the data
+    date_time_index_to_use = (
+        SATELLITE_DATETIME_INDEX if SATELLITE_DATA in required_keys else NWP_TARGET_TIME
+    )
+    current_time = batch[date_time_index_to_use][0, current_timestep_index]
```

Comment:
Just to be super-explicit about what's happening on this line, I'd be tempted to make use of `xarray`'s named dimensions and do something like:

```suggestion
    current_time = batch[date_time_index_to_use].isel(time=current_timestep_index)
```

(although, I must admit I haven't tested this, and I'm not 100% certain that the time dimension is called `time` in both `satellite_datetime_index` and `nwp_target_time`!)

Reply:
This is currently called after the xarray data is changed to Numpy arrays, but I try moving it futher up before the call to `example.to_numpy()`

Reply:
aha, sorry, got it!  Sorry I missed that!  I guess my other comments below also don't make sense given the data is now in numpy arrays!